_main_project/static/js/helpers/router.js

```diff
@@ -58,10 +58,27 @@ export const router = async () => {
   const view = new match.route.view(getParams(match));
 
   if (user.getIsInTournament() === true && view.name !== "TournamentLobby") {
-    // console.log('user should leave the tournament');
+    
+    // const leavingTournamentSocket = new WebSocket(`ws://${window.location.host}/ws/tournament_lobby/${tournamentID}/`);
+
+    // // Remove the player from the tournament
+    const tournamentURL = user.getTournamentURL();
+    const response = await fetch(`/tournament/remove_player_from_tournament/${tournamentURL}/`)
+  
+
+    const responseText = await response.text();
+    const data = JSON.parse(responseText);
+    
+    console.log('Data after leaving the tournament :', data);
+
+    if (data.status === 'error') {
+      console.error('Error while leaving tournament:', data.message);
+    }
+
+    user.setIsInTournament(false, '');
     user.getTournamentSocket().close();
-    user.setIsInTournament(false);
     user.setTournamentSocket(null);
+
   }
```

Comment:
I'm just going to leave a note suggesting some error checking on this code. I can come back to it later. but I dont know what happens if user.getTournamentSocket().close() fails. or if the fetch request fails. Since we have a JSON.parse right after, this can cause an error I think if an empty object is returned.
But like I said we can come back later to this